proxy/src/main/java/com/velocitypowered/proxy/connection/client/LoginSessionHandler.java

```diff
@@ -114,6 +115,12 @@ public boolean handle(EncryptionResponse packet) {
       throw new IllegalStateException("No EncryptionRequest packet sent yet.");
     }
 
+    if (receivedEncryptionResponse) {
+      throw new IllegalStateException("Too many EncryptionResponse packets received.");
```

Comment:
Also begs the question wether or not we should throw cached exceptions here

Reply:
potentially, idk how it will handle it if a flood of packets come in, will it throw on just the first or will it drain the others out? if it's just a single throw then it's not *too* bad, but, this is a stupidly easy area to send packets to, so, maybe much of this stuff should be cached exceptions?

Reply:
this should be a stackless exception